arch/arc/core/cache.c

```diff
@@ -93,8 +76,8 @@ static void dcache_flush_mlines(u32_t start_addr, u32_t size)
 		return;
 	}
 
-	end_addr = start_addr + size - 1;
-	start_addr &= (u32_t)(~(DCACHE_LINE_SIZE - 1));
+	end_addr = (u32_t)start_addr + size - 1;
+	*((u32_t *)start_addr) &= (u32_t)(~(DCACHE_LINE_SIZE - 1));
```

Comment:
This is writing *through* the pointer and storing data.  That can't be what you intend, can it?  The original code there was just aligning it downward to the next lower cache line.

Reply:
yes this isn't right, the purpose of that code is to align the address pointer

Reply:
My bad. Addressed this issue now. Thanks for pointing that out.